packages/desktop-client/e2e/budget.mobile.test.js

```diff
@@ -169,57 +213,125 @@ budgetTypes.forEach(budgetType => {
-      await budgetPage.waitForBudgetTable();
 
       const categoryName = await budgetPage.getCategoryNameForRow(0);
-      await budgetPage.openBudgetMenu(categoryName);
+      const budgetMenuModal = await budgetPage.openBudgetMenu(categoryName);
 
-      const budgetMenuModalHeading = page
-        .getByRole('dialog')
-        .getByRole('heading');
-
-      await expect(budgetMenuModalHeading).toHaveText(categoryName);
+      await expect(budgetMenuModal.heading).toHaveText(categoryName);
       await expect(page).toMatchThemeScreenshots();
     });
 
     test('updates the budgeted amount', async () => {
       const budgetPage = await navigation.goToBudgetPage();
-      await budgetPage.waitForBudgetTable();
 
       const categoryName = await budgetPage.getCategoryNameForRow(0);
+      const budgetMenuModal = await budgetPage.openBudgetMenu(categoryName);
+
+      const budgetAmount = 123;
 
-      // Set to 100.00
-      await budgetPage.setBudget(categoryName, 10000);
+      // Set to 123.00
+      await budgetMenuModal.setBudgetAmount(`${budgetAmount}00`);
 
       const budgetedButton =
         await budgetPage.getButtonForBudgeted(categoryName);
 
-      await expect(budgetedButton).toHaveText('100.00');
+      await expect(budgetedButton).toHaveText(amountToCurrency(budgetAmount));
+      await expect(page).toMatchThemeScreenshots();
+    });
+
+    test(`copies last month's budget`, async () => {
+      const budgetPage = await navigation.goToBudgetPage();
+
+      const categoryName = await budgetPage.getCategoryNameForRow(3);
+      const budgetedButton =
+        await budgetPage.getButtonForBudgeted(categoryName);
+
+      await budgetPage.goToPreviousMonth();
+
+      const lastMonthBudget = await budgetedButton.textContent();
+
+      await budgetPage.goToNextMonth();
+
+      await copyLastMonthBudget(budgetPage, categoryName);
+
+      await expect(budgetedButton).toHaveText(lastMonthBudget);
+      await expect(page).toMatchThemeScreenshots();
+    });
+
+    [
+      [3, setTo3MonthAverage],
+      [6, setTo6MonthAverage],
+      [12, setToYearlyAverage],
+    ].forEach(([numberOfMonths, setBudgetAverageFn]) => {
+      test(`set budget to ${numberOfMonths} month average`, async () => {
+        const budgetPage = await navigation.goToBudgetPage();
+
+        const categoryName = await budgetPage.getCategoryNameForRow(3);
+
+        const averageSpent = await setBudgetAverage(
+          budgetPage,
+          categoryName,
+          numberOfMonths,
+          setBudgetAverageFn,
+        );
+
+        const budgetedButton =
+          await budgetPage.getButtonForBudgeted(categoryName);
+
+        await expect(budgetedButton).toHaveText(
+          amountToCurrency(Math.abs(averageSpent)),
+        );
+        await expect(page).toMatchThemeScreenshots();
+      });
+    });
+
+    test(`applies budget template`, async () => {
+      const settingsPage = await navigation.goToSettingsPage();
```

Comment:
The pipeline's saying this is flaky: 

https://github.com/actualbudget/actual/actions/runs/11389100433/job/31687676786#step:5:397